share/smack/lib/intrinsics.c

```diff
@@ -0,0 +1,29 @@
+#include "smack.h"
+
+// Now this guard is unnecessary but if we decide to use this file
+// to implement all the intrinsics, we will need it.
+#if FLOAT_ENABLED
```

Comment:
In C++ code, this is guarded by both float and bit-precise being enabled. Here, it is guarded just by float. Why is there this discrepancy?

Reply:
The short answer is that we don't have a macro associated with the flag `--bit-precise`.  Currently we generate Boogie floating-point functions used in these C functions even when `--bit-precise` is not enabled. So we're good for now.